openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -652,6 +655,26 @@ def _apply_cdn_to_url(self, url, base_url):
 
         return urlunparse((None, base_url, path, params, query, fragment))
 
+    def get_prefetched_course_mode(self, mode_slug):
```

Comment:
@nasthagiri - I'm a little nervous about the fact that this contains a lot of business logic. Some of it is also present in `CourseMode` - however `CourseMode` performs this filtering in the database, which would prevent us from using the prefetched results.

Thoughts on this issue? Maybe move this logic into `CourseMode` with comments in both locations of similar logic?